WordPress/openverse-frontend
# Use vue-portal for VTeleport

## Fixes


Hopefully:
Fixes #1078 by @ sarayourfriend
Fixes #615 by @ obulat
Fixes #1417 by @ AetherUnbound 

## Description


This PR replaces the custom implementation of Teleport with `vue-portal` library that hopefully handles the SSR better. The library hasn't been updated for some period of time, but it's probably caused by the fact that Vue 2 is not the default anymore, and Vue3 has a native implementation of `Teleport`.

## Testing Instructions


Currently, if you look at the CI Playwright tests' log, you'll see a lot of warnings like these:
```
 playwright_1  |   ✓  e2e/filters.spec.ts:145:1 › filters are updated when media type changes (11s)
playwright_1  | [WebServer] › Rendering url /search/audio?q=birds&source=freesound
playwright_1  | [WebServer] 
playwright_1  |  WARN  VTeleportTarget: duplicate name skip-to-content, deleting previous teleport
playwright_1  | [WebServer] 
playwright_1  |  WARN  VTeleportTarget: duplicate name sidebar, deleting previous teleport
playwright_1  | [WebServer] 
playwright_1  |  WARN  VTeleportTarget: duplicate name modal, deleting previous teleport
```
The CI logs for this PR shouldn't have warnings about deleting portal targets, and the tests should not fail due to crashes.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
